### ISSUE TITLE
Qwen3.5-27B 训练速度极慢

### ISSUE THREAD

**Issue by justStarG** [modelscope/ms-swift#8096]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

使用 ms-swift 最新源码安装，最新版 transformers，训练 Qwen3.5-27B 速度极慢，而且越来越慢，CPU 基本利用率 100%，但是 GPU 空闲。我使用的 8 卡 H100，开的 deepspeed zero3。

Train:   0%|          | 1/3527 [00:15<12:16:56, 12.54s/it]
Train:   0%|          | 1/3527 [00:15<12:16:56, 12.54s/it]
Train:   0%|          | 2/3527 [00:31<15:45:18, 16.09s/it]
Train:   0%|          | 3/3527 [01:33<36:39:48, 37.45s/it]
Train:   0%|          | 4/3527 [02:50<51:51:42, 53.00s/it]
Train:   0%|          | 5/3527 [04:17<63:50:18, 65.25s/it]
                                                          
{'loss': '1.001', 'grad_norm': '1.015', 'learning_rate': '5e-07', 'token_acc': '0.7386', 'epoch': '0.001418', 'global_step/max_steps': '5/3527', 'elapsed_time': '4m 18s', 'remaining_time': '2d 2h 26m 53s', 'memory(GiB)': '39.34', 'train_speed(s/it)': '51.57'}

Train:   0%|          | 5/3527 [04:17<63:50:18, 65.25s/it]
Train:   0%|          | 5/3527 [04:17<63:50:18, 65.25s/it]
Train:   0%|          | 6/3527 [05:31<66:45:29, 68.26s/it]
Train:   0%|          | 7/3527 [06:45<68:23:17, 69.94s/it]
Train:   0%|          | 8/3527 [07:58<69:30:32, 71.11s/it]

### How to Reproduce / 如何复现

```
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=8 \
nohup swift sft \
--model /tmp/Qwen3.5-27B/ \
--dataset  \
train.jsonl \
--val_dataset val.jsonl  \
--train_type lora \
--torch_dtype bfloat16 \
--num_train_epochs 1 \
--per_device_train_batch_size 1 \
--per_device_eval_batch_size 1 \
--learning_rate 3e-5 \
--lora_rank 300 \
--lora_alpha 300 \
--freeze_llm false \
--freeze_vit true \
--gradient_accumulation_steps 1 \
--save_steps 500 \
--save_total_limit 5 \
--logging_steps 5 \
--max_new_tokens 12800 \
--max_length 14400 \
--output_dir outputs/ \
--warmup_steps 300 \
--dataloader_num_workers 0 \
--optim adamw_8bit \
--attn_impl flash_attn \
--do_eval false \
--truncation_strategy delete \
--torch_empty_cache_steps 10 \
--max_pixels 409600 --deepspeed zero3  > ./train.log 2>&1 &
```

### Additional Information / 补充信息

[WARNING] [engine.py:1522:_do_optimizer_sanity_check] **** You are using ZeRO with an untested optimizer, proceed with caution *****
Stage 3 initialize beginning
MA 10.51 GB         Max_MA 16.96 GB         CA 14.91 GB         Max_CA 17 GB 
CPU Virtual Memory:  used = 122.17 GB, percent = 6.5%
DeepSpeedZeRoOffload initialize [begin]
MA 10.51 GB         Max_MA 10.51 GB         CA 14.91 GB         Max_CA 15 GB 
CPU Virtual Memory:  used = 122.2 GB, percent = 6.5%
Parameter Offload - Persistent parameters statistics: param_count = 670, numel = 4437232
DeepSpeedZeRoOffload initialize [end]
MA 6.92 GB         Max_MA 10.51 GB         CA 15.17 GB         Max_CA 15 GB 
CPU Virtual Memory:  used = 122.82 GB, percent = 6.6%
Before creating fp16 partitions
MA 6.92 GB         Max_MA 6.92 GB         CA 15.17 GB         Max_CA 15 GB 
CPU Virtual Memory:  used = 122.82 GB, percent = 6.6%
After creating fp16 partitions: 1
MA 6.9 GB         Max_MA 6.92 GB         CA 7.27 GB         Max_CA 15 GB 
CPU Virtual Memory:  used = 122.85 GB, percent = 6.6%
Before creating fp32 partitions
MA 6.9 GB         Max_MA 6.9 GB         CA 7.27 GB         Max_CA 7 GB 
CPU Virtual Memory:  used = 122.84 GB, percent = 6.6%
After creating fp32 partitions
MA 7.92 GB         Max_MA 8.94 GB         CA 9.3 GB         Max_CA 9 GB 
CPU Virtual Memory:  used = 122.85 GB, percent = 6.6%
Before initializing optimizer states
MA 7.92 GB         Max_MA 7.92 GB         CA 9.3 GB         Max_CA 9 GB 
CPU Virtual Memory:  used = 122.87 GB, percent = 6.6%
After initializing optimizer states
MA 7.92 GB         Max_MA 8.94 GB         CA 9.3 GB         Max_CA 9 GB 
CPU Virtual Memory:  used = 122.85 GB, percent = 6.6%
After initializing ZeRO optimizer
MA 8.47 GB         Max_MA 8.48 GB         CA 9.3 GB         Max_CA 9 GB 
CPU Virtual Memory:  used = 122.89 GB, percent = 6.6%

**Comment by HQR2000** [comment]
遇到类似的问题，ms-swift 最新源码安装，单张H20上微调Qwen3.5-27B，速度也是非常慢

Train:   0%|          | 1/8532 [01:10<167:02:42, 70.49s/it]
                                                           

Train:   0%|          | 1/8532 [01:10<167:02:42, 70.49s/it]
Train:   0%|          | 1/8532 [01:10<167:02:42, 70.49s/it]
Train:   0%|          | 2/8532 [02:18<163:27:10, 68.98s/it]
Train:   0%|          | 3/8532 [03:17<152:20:51, 64.30s/it]
Train:   0%|          | 4/8532 [04:24<154:45:56, 65.33s/it]
Train:   0%|          | 5/8532 [05:46<169:13:48, 71.45s/it]
Train:   0%|          | 6/8532 [07:11<180:18:44, 76.13s/it]
Train:   0%|          | 7/8532 [08:38<188:15:44, 79.50s/it]
Train:   0%|          | 8/8532 [09:48<181:38:42, 76.72s/it]
Train:   0%|          | 9/8532 [10:59<177:20:00, 74.90s/it]
Train:   0%|          | 10/8532 [12:28<187:27:06, 79.19s/it]
                                                            

Train:   0%|          | 10/8532 [12:28<187:27:06, 79.19s/it]
Train:   0%|          | 10/8532 [12:28<187:27:06, 79.19s/it]
Train:   0%|          | 11/8532 [13:43<184:08:06, 77.79s/it]

### 启动命令
`
swift sft \
  --model models--Qwen--Qwen3.5-27B/snapshots/b7ca741b86de18df552fd2cc952861e04621a4bd \
  --dataset train.jsonl \
  --tuner_type lora \
  --lora_rank 8 \
  --lora_dropout 0.05 \
  --torch_dtype bfloat16 \
  --freeze_vit true \
  --num_train_epochs 2 \
  --per_device_train_batch_size 4 \
  --per_device_eval_batch_size 1 \
  --gradient_accumulation_steps 1 \
  --learning_rate 2e-5 \
  --lr_scheduler_type cosine \
  --output_dir outputs/ \
  --save_steps 50 \
  --logging_steps 10 \
  --warmup_ratio 0.1 \
  --dataset_num_proc 16 \
  --dataloader_num_workers 16
`

**Comment by justStarG** [comment]
找到问题了，可能是 deepspeed 默认配置有问题，offload 导致速度变慢了很多。使用这个配置速度就快很多了，一个 iter大概7.9秒。

```json
{
    "bf16": {
        "enabled": true
    },
    "zero_optimization": {
        "stage": 3,
        "offload_optimizer": {
            "device": "none"
        },
        "offload_param": {
            "device": "none"
        },
        "overlap_comm": true,
        "contiguous_gradients": true,
        "sub_group_size": 1e9,
        "reduce_bucket_size": "auto",
        "stage3_prefetch_bucket_size": "auto",
        "stage3_param_persistence_threshold": "auto",
        "stage3_max_live_parameters": 1e9,
        "stage3_max_reuse_distance": 1e9,
        "stage3_gather_16bit_weights_on_model_save": true
    },
    "gradient_accumulation_steps": "auto",
    "gradient_clipping": "auto",
    "steps_per_print": 100,
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "wall_clock_breakdown": false
}

```

**Comment by maksimlitvinov39kg** [comment]
@justStarG Hello, can you please share your env settings/ pip list, which you used for qwen3.5 finetuning?

**Comment by Jintao-Huang** [comment]
<img width="659" height="154" alt="Image" src="https://github.com/user-attachments/assets/17e2dcd5-3063-417d-8d1a-1ef5c761c0e2" />

**Comment by justStarG** [comment]
> [@justStarG](https://github.com/justStarG) Hello, can you please share your env settings/ pip list, which you used for qwen3.5 finetuning?

swift 用的源码，模型是`Qwen3.5-27B`，torch 是 2.10

**Comment by zkailinzhang** [comment]
想问下，train.jsonl 是vqa数据吗  样式是什么样的

**Comment by justStarG** [comment]
多模态的训练数据，有 prompt、图片和结果。

**Comment by zkailinzhang** [comment]
> 多模态的训练数据，有 prompt、图片和结果。

类似这样吗
{"messages": [{"role": "user", "content": "两张图片有什么区别"}, {"role": "assistant", "content": "前一张是小猫，后一张是小狗"}], "images": ["/xxx/x.jpg", "/xxx/x.png"]}
